google/cloud/storage/client.h

```diff
@@ -3291,16 +3291,16 @@ class Client {
       " Please file a bug at https://github.com/googleapis/google-cloud-cpp"
       " if you have a use-case not covered by these.")
 #endif  // _MSC_VER
-  // NOLINTNEXTLINE(performance-unnecessary-value-param)
-  explicit Client(std::shared_ptr<internal::RawClient> client,
+  explicit Client(std::shared_ptr<internal::RawClient> const& client,
```

Comment:
`shared_ptr<T> const& param` smells funny.

Reply:
It does, added a comment explaining why.